### DISCUSSION TITLE
How to set debounce on $wire.entangle

### DISCUSSION THREAD

**Discussion by ariaieboy** [livewire/livewire#7109] [1. Help]
How can we set the debounce amount in $wire.entangle?

**Comment by satoved** [comment]
You don't set it in entangle itself, you wrap value synchronization with a custom debounce.

E.g. using lodash https://lodash.com/docs/4.17.15#debounce

You can then have two properties inside x-data - one real time with x-model on it, and another one entangled.

Then sync them with watch or events are wrap the synchronization function in debounce closure.

**Comment by Muffinman** [comment]
Alpine has it's own native debounce you can use, I did something similar recently but I could only do it having a separate Alpine variable. I'm sure someone cleverer could do this in a one-liner.

```blade
<div class="relative flex justify-between {{ $this->class ?? null }}" x-data="{
    quantity: @entangle('quantity').live,
    tempQuantity: {{ $quantity }},

    init () {
        $watch('tempQuantity', Alpine.debounce(() => this.quantity = this.tempQuantity, 200))
    },
    increase() {
        this.tempQuantity++
    },
    decrease() {
        this.tempQuantity--
        if (this.tempQuantity < 0) {
            this.tempQuantity = 0
        }
    }
}">
    <div class="flex items-center">
        <span x-on:click.prevent="decrease" class="flex px-1 py-1 items-center cursor-pointer">
            <x-icons.minus />
        </span>
        <label for="quantity" class="sr-only">Quantity</label>
        <input
            x-model="tempQuantity"
            type="number"
            step="1"
            min="0"
            name="tempQuantity"
            class="w-8 min-w-0 text-center text-sm bg-transparent"
            />
        <span x-on:click.prevent="increase" class="flex px-1 py-1 items-center cursor-pointer">
            <x-icons.plus />
        </span>
    </div>

    <input
        x-model="quantity"
        type="number"
        step="1"
        min="0"
        name="quantity"
        class="hidden"
    />
</div>


```

**Reply by ariaieboy** [reply]
@satoved First of all, I don't want to install any node dependency, and since we have the same API with `wire:model` I think we should have a similar API with entangle
@Muffinman, your implementation is good, but if you read the first comment of the debounce feature in livewire here  https://github.com/livewire/livewire/blob/main/js/debounce.js, there are lots of edge cases here.
What if the user presses the submit button before the entangle gets updated by the debounces method?

**Reply by Muffinman** [reply]
Checking the source for `Blade::directive('entangle')` it doesn't like there are any other params for specifying the delay (that I can see).

In my case the user doesn't have any other submit buttons because it's a simple quantity increase. If state is critical I guess you could use Alpine to disable other actions on the form after while the debounce is in progress.

**Reply by ariaieboy** [reply]
The problem is that, in my scenario, I do not have control over the form.
And it's a missing feature for Livewire.

**Comment by bennettblack** [comment]
This can be achieved by debouncing the model binding, and applying `.live` to your entanglement.

```blade
<div x-data="{ 
    searchQuery: $wire.entangle('searchQuery').live,
    }"
    >
    <input type="text"
        x-model.debounce.750ms="searchQuery"
        />
</div>
```

**Comment by dayemsiddiqui** [comment]
Hi I was able to achieve a debounced sync using the following one liner:

```
  <span wire:dirty wire:poll.visible="syncInvoice"
          class="opacity-0 -z-10 fixed w-screen h-screen">Syncing Invoice</span>
```

It combines the following two aspects :

> https://livewire.laravel.com/docs/wire-poll#viewport-throttling
> Viewport throttling
> Another measure you can take to only poll when necessary, is to add the .visible modifier to wire:poll. The .visible modifier instructs Livewire to only poll the component when it is visible on the page:
> 
> <div wire:poll.visible>
> If a component using wire:visible is at the bottom of a long page, it won't start polling until the user scrolls it into the viewport. When the user scrolls away, it will stop polling again.

And this:

> https://livewire.laravel.com/docs/wire-dirty#basic-usage
> By adding wire:dirty to an element, you are instructing Livewire to only show the element when the client-side state diverges from the server-side state.
